test/libp2p.spec.js

```diff
@@ -0,0 +1,431 @@
+  })
+
+  it('libp2p.hangupByPeerInfo nodeA to nodeB', (done) => {
+    nodeA.hangUpByPeerInfo(nodeB.peerInfo, (err) => {
+      expect(err).to.not.exist
+      setTimeout(check, 500)
+
+      function check () {
+        parallel([
+          (cb) => {
+            const peers = nodeA.peerBook.getAll()
+            expect(err).to.not.exist
+            expect(Object.keys(peers)).to.have.length(0)
+            expect(Object.keys(nodeA.swarm.muxedConns)).to.have.length(0)
+
+            cb()
+          },
+          (cb) => {
+            const peers = nodeB.peerBook.getAll()
+            expect(err).to.not.exist
+            expect(Object.keys(peers)).to.have.length(0)
+            expect(Object.keys(nodeB.swarm.muxedConns)).to.have.length(0)
+            cb()
+          }
+        ], done)
+      }
+    })
+  })
+
+  // NOTE, these dialById only works if a previous dial was made
+  // until we have PeerRouting
+  it.skip('libp2p.dialById nodeA to nodeB', (done) => {})
+  it.skip('libp2p.dialById on Protocol nodeA to nodeB', (done) => {})
+  it.skip('libp2p.hangupById nodeA to nodeB', (done) => {})
+
+  it('nodeA dial to nodeC and nodeD', (done) => {
+    let count = 0
+    const next = () => ++count === 2 ? check() : null
+
+    nodeA.dialByMultiaddr(nodeCMultiaddrTCP, next)
+    nodeA.dialByMultiaddr(nodeDMultiaddrTCP, next)
+
+    function check () {
+      const peers = nodeA.peerBook.getAll()
+      expect(Object.keys(peers)).to.have.length(2)
+      done()
+    }
+  })
+
+  // Multiple transport tests
+
+  it('nodeA dial to nodeE', (done) => {
+    nodeA.dialByMultiaddr(nodeEMultiaddrTCP, (err) => {
+      expect(err).to.not.exist
+      const peers = nodeA.peerBook.getAll()
+      expect(Object.keys(peers)).to.have.length(3)
+      done()
+    })
+  })
+
+  it('nodeA fails to dial to nodeF', (done) => {
+    nodeA.dialByMultiaddr(nodeFMultiaddrWebSockets, (err) => {
+      expect(err).to.exist
+      const peers = nodeA.peerBook.getAll()
+      expect(Object.keys(peers)).to.have.length(3)
+      done()
+    })
+  })
+
+  it('nodeF fails to dial to nodeA', (done) => {
+    nodeF.dialByMultiaddr(nodeAMultiaddrTCP, (err) => {
+      expect(err).to.exist
+      done()
+    })
+  })
+
+  it('nodeF dial to nodeE', (done) => {
+    nodeF.dialByMultiaddr(nodeEMultiaddrWebSockets, (err) => {
+      expect(err).to.not.exist
+      done()
+    })
+  })
+
+  // Turbolence tests
+
+  it('spawn a node in a different process', (done) => {
+    const filePath = path.join(__dirname, './spawn-libp2p-node.js')
+    spawnedNode = spawn(filePath)
+
+    let spawned = false
+    spawnedNode.stdout.on('data', (data) => {
+      // console.log(`stdout: ${data}`)
+      if (!spawned) {
+        spawned = true
+        done()
+      }
+    })
+
+    spawnedNode.stderr.on('data', (data) => {
+      // console.log(`stderr: ${data}`)
+    })
+
+    spawnedNode.on('close', (code) => {
+      // console.log(`child process exited with code ${code}`)
+    })
+  })
```

Comment:
This test spawns  libp2pNode in a different process, which gets killed in the next test. I used this event listeners for debugging and left them here. I can remove it :)